lib/db/mysql.js

```diff
@@ -566,21 +566,31 @@ module.exports = function (log, error) {
 
   // Internal
 
-  MySql.prototype.singleQuery = function (poolName, sql, params) {
+  MySql.prototype.singleQuery = function (poolName, sql, params, readUncommitted) {
     return this.getConnection(poolName)
       .then(
         function (connection) {
-          return query(connection, sql, params)
-            .then(
-              function (result) {
-                connection.release()
-                return result
-              },
-              function (err) {
-                connection.release()
-                throw err
-              }
-            )
+          if (readUncommitted) {
+            return query(connection, 'SET TRANSACTION ISOLATION LEVEL READ UNCOMMITTED')
```

Comment:
TBH those docs are not entirely clear to me.  Without `GLOBAL` or `SESSION`, do you have to explicitly start a transaction for the isolation level to take effect?  Or does a subsequent statement count as an implicit transaction?


My reflex would be to run them all in a single, explicit transaction:

```
    SET TRANSACTION ISOLATION LEVEL ...
    START TRANSACTION
    metrics query
    metrics query
    ...
    END TRANSACTION
```

But you'd still have the issue of ensuring the transaction is properly torn down in case of error, which would require further surgery to the db lib API.

Trying it out, it does appear to apply correctly to the subsequent statement and then revert.  I did this on one session:

```
mysql> create table test1 (id VARCHAR(256) NOT NULL);
mysql> START TRANSACTION;
mysql> INSERT INTO test1 VALUES ("hello");
mysql> -- leave the transaction open...
```

And then this on another:

```
mysql> select * from test1;
Empty set (0.00 sec)

mysql> SET TRANSACTION ISOLATION LEVEL READ UNCOMMITTED;
Query OK, 0 rows affected (0.00 sec)

mysql> select * from test1;
+-------+
| id    |
+-------+
| hello |
+-------+
1 row in set (0.00 sec)

mysql> select * from test1;
Empty set (0.00 sec)
```

So I think you're right, and this:

> simply use SET TRANSACTION ISOLATION LEVEL... before each query and
> have the isolation revert to the session default automatically.

is probably going to be the best way to go.